lib/active_merchant/billing/gateways/agms.rb

```diff
@@ -0,0 +1,322 @@
+# Agms Gateway uses nokogiri to build the xml request
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class AgmsGateway < Gateway
+      self.test_url = 'https://gateway.agms.com/roxapi/agms.asmx'
```

Comment:
Test vs. Live is configured in the gateway account itself.  The credentials we are providing are for a gateway account configured for test mode, both do use the same endpoints.